st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -92,6 +125,50 @@ def _get_default(_actionchain):
         return _actionchain.chain[0].name
 
     @staticmethod
+    def _get_all_nodes(action_chain):
```

Comment:
Theoretical argument aka feel free to ignore -

The fact that an ActionChain is valid i.e. the `on-success` and `on-failure` links are broken or how to interpret the broken-ness is a function of the chainrunner. It is possible for a different implementation of the ActionChainRunner to interpret this condition differently. Therefore, validity of an ActionChain in this context can only be decided by this runner.

In the context of StackStorm this argument is mostly invalid since we would not be writing another runner anyway therefore theoretical. However, since we build software with certain recognizable techniques (aka patterns) and hope that the next new person who shows up can semi-intuitively understand the recognizable patterns and follow along I feel it is worthwhile to be consistent about such things.

As such I rest my case and if you do prefer to move it out to ActionChain model itself go for it :)